server/collection/permissions.ts

```diff
@@ -1,6 +1,27 @@
 import { getScope } from 'server/utils/queryHelpers';
 
-export const getPermissions = async ({ userId, communityId, collectionId }) => {
+export const collectionUpdatePremission = [
+	'title',
+	'slug',
+	'isRestricted',
+	'isPublic',
+	'pageId',
+	'metadata',
+	'readNextPreviewSize',
+	'layout',
+	'layoutAllowsDuplicatePubs',
+	'avatar',
+] as const;
+
+export const getPermissions = async ({
+	userId = null,
+	communityId = null,
+	collectionId = null,
+}: {
+	userId: string | null;
+	communityId: string | null;
+	collectionId: string | null;
+}): Promise<Permissions> => {
 	if (!userId) {
 		return {};
```

Comment:
This is something I do a number of times, explicitly type the permissions.

This was not to prevent any type errors, if anything it introduced more, but this proves to be useful when typing the queries more accurately, which is something we want to do eventually.